src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -855,10 +884,23 @@ private static string FormatDiagnostics(ImmutableArray<DiagnosticAnalyzer> analy
                 }
                 else
                 {
-                    AppendLocation(diagnostics[i].Location);
-                    foreach (var additionalLocation in diagnostics[i].AdditionalLocations)
+                    // The unnecessary code designator is ignored for the primary diagnostic location.
+                    AppendLocation(diagnostics[i].Location, isUnnecessary: false);
+
+                    int[] unnecessaryIndices = { };
+                    if (diagnostics[i].Properties.TryGetValue(WellKnownDiagnosticTags.Unnecessary, out var encodedUnnecessaryLocations))
+                    {
+                        var match = EncodedIndicesSyntax.Match(encodedUnnecessaryLocations);
+                        if (match.Success)
+                        {
+                            unnecessaryIndices = match.Groups["Index"].Captures.OfType<Capture>().Select(capture => int.Parse(capture.Value)).ToArray();
```

Comment:
>ToArray [](http://example.com/codeflow?start=142&length=7)

perf doesn't matter here, but this is used more as a set than an array.

Reply:
➡️ Will keep this form, since the collection is typically small, and if we ever want to add assertions about uniqueness or order it will be clearer what we are starting with.